fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/FabricItemUpdateAnimationHandlers.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.client.ItemUpdateAnimationHandlerExtensions;
+
+@Environment(EnvType.CLIENT)
+public class FabricItemUpdateAnimationHandlers {
```

Comment:
Honestly not too sure why this exist when one could implement the `UpdateAnimationHandler` or use mixins to implement it on a vanilla item.

Also this allows other mods to override a mods implementation ( if they choose to implement the interface ) are you sure this is a good idea?

Otherwise PR looks good.

Reply:
Just to clarify, this was originally an interface you could implement on items, but that was vetoed in favor of a registry. The interface is now only used as a functional interface (I should probably add the annotation to it).

The current "registry"-based system chucks out an error when you try to override other mods handlers on the same item. I do think there is a use case for letting multiple handlers act on the same item, but as I said in an earlier comment, that would basically just turn into a standard event (which we could entirely do if needed).

Reply:
Oh looks like I misread some more of the code, my bad. PR looks good to me, In my previous use case I only ever needed this for my own items.